### PR TITLE
Remove useFcmV1 Due to Upcoming Legacy API Deprecation

### PR DESCRIPTION
**Pull request by katayama8000**
We don't need the useFcmV1 because the legacy API won't be available after June 20, 2024.
https://expo.dev/blog/expo-adds-support-for-fcm-http-v1-api#fcm-timeline

### CODE REVIEW

**Comment by edi** [comment]
You can't just remove it because people who use TS and have that param set will now have issues compiling. This parameter can still be sent by the user whilst the backend will simply ignore it after the 20th of June, since it will default to FCM V1 Key (if present).

A deprecation warning while the types are still supported is recommended.

Regards

**Comment by katayama8000** [comment]
@edi 
I got it.
But Is it possible to indicate that an argument is deprecated?

**Comment by edi** [comment]
@katayama8000 Yes, you can. Check the following JSDoc resource

https://www.typescriptlang.org/play/?#example/jsdoc-deprecated

**Comment by katayama8000** [comment]
@edi 
Is it what you want?
We do not refer to useFcmV1.
So We want users to know that useFcmV1 is deprecated when instantiate right?

**Comment by katayama8000** [comment]
@edi 
https://www.typescriptlang.org/play/?#code/MYGwhgzhAEBiD28DeAoa7oHoBU20em2gAEATAUwAcAnc4MAF3NOnhBYDMBXAO2AYCW8HvgzZMo6MGEQG1Lv3jUAFGABcs6gJ4BzADTQARhrnb9UtYcQhyYHgEoA3JJx4ChaD3IB3aNz6CwpKEEgTSPJoKDEqqJlq6BsaaZk6S4ZGKKurJCUZxZgbAAPyW1rYOqJIAvig1KChevgjwygBEYK16rYatqY1wiG0dXT16clzkTkA
This is my idea.

**Comment by edi** [comment]
@katayama8000 Problem with that is people will get confused, thinking that the whole Expo client is deprecated, as most don't actually understand JSDoc.

So I'll stick with my initial suggestion of marking `useFcmV1` as deprecated, adding a note explaining why it is deprecated, and maybe linking  it [here](https://expo.dev/blog/expo-adds-support-for-fcm-http-v1-api#fcm-timeline) as well

**Comment by katayama8000** [comment]
@edi Thank you for the suggestion!

**Comment by katayama8000** [comment]
@ide I wanna ask your opinion.

File: package.json
Comment (ide) [inline package.json:3]: In general, avoid bumping versions in PRs. Package maintainers will bump the versions before publishing.

**Review by ide — COMMENTED** [review]
This looks fine to me, aside from the package.json version change. We'll probably want to publish this as a new major version because it's technically a breaking change.

Google is shutting down FCM Legacy on June 21, so let's hold off on merging this until then.

File: package.json
Comment (katayama8000) [reply package.json:3]: OK.

Comment (katayama8000) [reply package.json]: I just just deleted the commit.

**Comment by katayama8000** [comment]
@ide 
Thank you for your opinion!!!
I want you to review it.

**Review by ide — APPROVED** [review]

**Comment by ide** [comment]
Google extended the deadline by one month very recently. Let's revisit this a month from now.

**Comment by katayama8000** [comment]
@ide I think you can merge it.

**Comment by ide** [comment]
We are still seeing the legacy FCM service accept push notification requests and reply with successful statuses. This [page](https://firebase.google.com/support/faq#deprecated-api-shutdown) says the shutdown is gradual. There is no rush from Expo to stop supporting the FCM legacy service for a little while longer, especially since Google indicated that requesting an extension will only prevent their flickering brownouts but will still be subject to the same hard cutoff, whenever that is.

**Comment by douglowder** [comment]
Closing this -- superseded by #76

**Pull request closed without merging**
